crates/cli/src/bytecode.rs

```diff
@@ -5,18 +5,21 @@ use wasmtime::{AsContextMut, Engine, Instance, Linker, Memory, Module, Store};
 pub const QUICKJS_PROVIDER_MODULE: &[u8] =
     include_bytes!(concat!(env!("OUT_DIR"), "/provider.wasm"));
 
-pub fn compile_source(js_source_code: &[u8]) -> Result<Vec<u8>> {
-    let (mut store, instance, memory) = create_wasm_env()?;
+/// Use the legacy provider when using the `compile -d` command.
+pub const QUICKJS_PROVIDER_V2_MODULE: &[u8] = include_bytes!("./javy_quickjs_provider_v2.wasm");
+
+pub fn compile_source(provider: &[u8], js_source_code: &[u8]) -> Result<Vec<u8>> {
+    let (mut store, instance, memory) = create_wasm_env(provider)?;
     let (js_src_ptr, js_src_len) =
         copy_source_code_into_instance(js_source_code, store.as_context_mut(), &instance, &memory)?;
     let ret_ptr = call_compile(js_src_ptr, js_src_len, store.as_context_mut(), &instance)?;
     let bytecode = copy_bytecode_from_instance(ret_ptr, store.as_context_mut(), &memory)?;
     Ok(bytecode)
 }
 
-fn create_wasm_env() -> Result<(Store<WasiCtx>, Instance, Memory)> {
+fn create_wasm_env(bytes: &[u8]) -> Result<(Store<WasiCtx>, Instance, Memory)> {
```

Comment:
Can we rename the argument to something like `provider_bytes`? Or just something to describe what the bytes are.